src/dex/camelot/camelot.ts

```diff
@@ -729,21 +717,22 @@ export class Camelot
     destToken: Address,
     srcAmount: NumberAsString,
     toAmount: NumberAsString, // required for buy case
-    data: UniswapData,
+    data: SolidlyData,
     side: SwapSide,
   ): AdapterExchangeParam {
     if (side === SwapSide.BUY) throw new Error('Buy not supported');
 
     const pools = encodePools(data.pools, this.feeFactor);
-    const weth = this.getWETHAddress(srcToken, destToken, data.weth);
+    const weth = this.getWETHAddress(srcToken, destToken, data.wethAddress);
     const payload = this.abiCoder.encodeParameter(
       {
         ParentStruct: {
           weth: 'address',
           pools: 'uint256[]',
+          isFeeTokenInRoute: 'bool',
         },
       },
-      { pools, weth },
+      { pools, weth, isFeeTokenInRoute: data.isFeeTokenInRoute },
```

Comment:
because this was missing, decoding was completely erroneous

Reply:
as the isFeeTokenInRoute bit was confused with the next bit which lead to take a different contract path